stablehlo/reference/Ops.cpp

```diff
@@ -769,6 +781,60 @@ Tensor evalFloorOp(const Tensor &operand, ShapedType resultType) {
   return result;
 }
 
+Tensor evalGatherOp(const Tensor &operand, const Tensor &startIndices,
+                    const Axes &offsetDims, const Axes &collapsedSliceDims,
+                    const Axes &startIndexMap, Axis indexVectorDim,
+                    const Sizes &sliceSizes, bool indicesAreSorted,
+                    ShapedType resultType) {
+  Tensor result(resultType);
+  Axes batchDims;
+  for (auto i = 0; i < result.getRank(); ++i)
+    if (!llvm::is_contained(offsetDims, i)) batchDims.push_back(i);
+
+  for (auto resultIt = result.index_begin(); resultIt != result.index_end();
+       ++resultIt) {
+    Index batchIndex;
+    for (auto batchDim : batchDims) batchIndex.push_back((*resultIt)[batchDim]);
+
+    Index startIndex;
+    Sizes batchIndexOffset(startIndices.getRank(), 1);
+    if (indexVectorDim < startIndices.getRank()) {
+      batchIndex.insert(batchIndex.begin() + indexVectorDim, 0);
+      batchIndexOffset[indexVectorDim] =
+          startIndices.getShape()[indexVectorDim];
+    }
```

Comment:
I am not sure if i get this. 

from the POV of slice op, the relation between `batch_index` and `limit_indices` (for the slice) is:
```
If batchIndex = [bi0, ..., bik, ..., biN] for k = index_vector_dims
then limit_indices = [bi0, ..., bik + dim(start_indices, index_vector_dims) -1, ..., biN]
```

A few concerns:
1.  why are we making the index value at `indexVectorDim` zero.
2.  Can we avoid the update of `batchIndex`?

Reply:
Ah, sorry I got confused here. Thanks!